app/controllers/transactions/categories/deletions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Transactions::Categories::DeletionsController < ApplicationController
+  before_action :set_category
+  before_action :set_replacement_category, only: :create
+
+  def new
+  end
+
+  def create
+    @category.reassign_transactions_then_destroy! @replacement_category
```

Comment:
This could potentially represent hundreds of transactions, is it too much of an early optimization to consider a background job to handle the bulk reassignment?

Reply:
Good callout! The bulk reassignment should be able to quickly handle even thousands of transactions at once without much problem (assuming a reasonably robust setup).

By performing this asynchronously, we'd be guarding against the request taking too long. That's fine! But at that scale I'd be more concerned about db throughput, not latency. Which means the actual fix would be batching the updates in addition to performing them async.

So the question becomes is it too early to batch? I think so. It's too defensive, too early. Suddenly we're deferring, batching, and throttling, and the product isn't even live yet. Which means requirements are likely to change, so there's more code to change when they do.

But people may reasonably disagree because this _will_ break with enough scale, even if it's a while before we reach it (recall we're not touching all transactions, but just the ones for this category). And the overhead is small.

Ultimately up to the maintainers how defensive they want to be, so let me know @zachgoll!

Reply:
While certainly not a bad thought, I think @josefarias is correct that we don't need to worry about this optimization yet.  Our main priority is to get features built as fast as possible and take a "Just in time" approach to optimizations rather than "Just in case".